Ginger/GingerCoreNET/RunLib/CLILib/CLIOptions.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System;
+using System.Reflection;
+using CommandLine;
+using GingerCoreNET.SourceControl;
+
+namespace Amdocs.Ginger.CoreNET.RunLib.CLILib
+{
+
+    class CLIOptionClassHelper
+    {
+        public const string FILENAME = "filename";
+
+        public static string GetClassVerb<T>()
+        {
+            VerbAttribute verbAttr = typeof(T).GetCustomAttribute<VerbAttribute>();
+            return verbAttr.Name;
+        }
+
+        public static string GetAttrShorName<T>(string name)
+        {
+            OptionAttribute attrOption = typeof(T).GetProperty(name).GetCustomAttribute<OptionAttribute>();
+            return attrOption.ShortName;
+        }
+    }
+
+    public class OptionsBase  // ' Base class for Options whihc execute run set
+    {
+        public enum eVerboseLevel
+        {
+            normal,
+            debug            
+        }
+
+        [Option('v', "verbose", Required = false,Default = eVerboseLevel.normal,  HelpText = "Select Verbose level: normal, debug")]
+        public eVerboseLevel VerboseLevel { get; set; }
+    }
+
+        
+
+    [Verb("example", HelpText = "Show example")]
+    public class ExampleOptions  // 'ginger example -v run' display CLI examples for run
+    {        
+        [Option('b', "verb", Required = true, HelpText = "Select Verb to show example")]
+        public string verb { get; set; }
+    }
+
+    [Verb("version", HelpText = "Show version information")]
+    public class VersionOptions  // 'ginger version
+    {
+        
+    }
+
+
+    [Verb("run", HelpText = "execute run set")]
+    public class RunOptions : OptionsBase
+    {
+        public static string Verb => CLIOptionClassHelper.GetClassVerb<RunOptions>();
+
+        [Option('s', "solution", Required = true, HelpText = "Set solution folder")]
+        public string Solution { get; set; }
+
+
+        [Option('r', "runset", Default = "Default Run Set", Required = true, HelpText = "Set runset name")]
+        public string Runset { get; set; }
+
+        [Option('e', "env",  Required = true, HelpText = "Set environment name")]
```

Comment:
we said env won't be mandatory because not all users using environments but it set as Required=true

Reply:
you so this comment?